code/scripting/api/libs/hud.cpp

```diff
@@ -53,6 +53,24 @@ ADE_VIRTVAR(HUDDisabledExceptMessages, l_HUD, "boolean", "Specifies if only the
 		return ADE_RETURN_FALSE;
 }
 
+ADE_VIRTVAR(HUDDefaultGaugeCount, l_HUD, "number", "Specifies the amount of HUD gauges defined by FSO", "number", "The number of FSO HUD gauges")
```

Comment:
I do not know what this is supposed to do. Could you elaborate on the use case for this?

Reply:
FSO defines a number of gauges, hardcoded, that are then rendered. This vector is, at the moment, expected to have 42 gauges, the HUD Config has 39 (I am not sure where the difference comes from, but hudparse.cpp defines 42 and hudgauges.h defines 39. I assume that hudparse has some gauges that were added over the years).
This virtvar returns the number of actually loaded gauges, so that in case a script needs to iterate over all existing gauges, it can do this with this variable, even if there may be further additions to the gauges in the future (as apparently, there have been). If this is guaranteed never to happen, this virtvar is unnecessary (then it should be noted in the docs that gauges range from index 0 to 41).

Reply:
Ok, then this is not correct thus far.

1. You need to return the amount like this below: `return ade_set_args(L, "i", amount);` There should be other functions that use something similar so you can use that for reference.
2. Since this is effectively a "read-only" variable you can omit the `ADE_SETTING_VAR` part since there is no meaningful implementation for that anyway.

Reply:
Alright, changed it.